Source/WebKit/Platform/IPC/StreamClientConnection.h

```diff
@@ -182,6 +184,19 @@ bool StreamClientConnection::send(T&& message, ObjectIdentifier<U> destinationID
     return true;
 }
 
+template<typename T, typename U, typename C>
+uint64_t StreamClientConnection::sendWithAsyncReply(T&& message, ObjectIdentifier<U> destinationID, Timeout timeout, C&& callback)
+{
+    static_assert(!T::isSync, "Message is sync!");
+    if (!trySendDestinationIDIfNeeded(destinationID.toUInt64(), timeout))
+        return 0;
+    auto span = tryAcquire(timeout);
```

Comment:
I think the main use-case would still be that the message would be fast to send, e.g. that the default case would send it as a stream message business as usual and then fallback case as out of stream message, business as usual.